### DISCUSSION TITLE
redirect/change url without refresh

### DISCUSSION THREAD

**Discussion by hamidrabedi** [django-commons/django-unicorn#591] [Ideas]
It would be great if you could add `pushState` and `replaceState` to achieve changing the URL without refreshing the page.

**Comment by adamghill** [comment]
Have you checked out https://www.django-unicorn.com/docs/redirecting/#locationupdate? Or does that not work for your use case?

**Reply by hamidrabedi** [reply]
oh yeah... i don't know why i didn't see that. tnx

**Comment by hamidrabedi** [comment]
Also, there are some other refactoring and feature requests:

One of the best use cases for Unicorn is for forms, but I can't customize it the way Unicorn operates. I had to write the entire thing myself, and it would be great to have the option to save, validate, and retrieve the form with the possibility of overriding the functions. Additionally, getting the fields dynamically from the form_class would be helpful.

Some other common use cases include pagination and filters.

There could be many improvements in handling exceptions as well. Syntax or logic errors do not always raise meaningful errors, making it challenging to track down the problem.

Thanks for your great work !

**Reply by adamghill** [reply]
For the forms support, I would appreciate some specific issues created with sample code of what you would _like_ for things to work. Personally, I don't use forms all that often so I do not run into the same sharp edges, I'm guessing.

>Syntax or logic errors do not always raise meaningful errors

Feel free to make issues that demonstrates these problems so I can replicate or PRs that provide a better DX.

Thanks!
